File: tests/algorithms/x/Java/ciphers/mixed_keyword_cypher.java
public class Main {
    static String UPPER;
    static String LOWER;

    static String to_upper(String s) {
        String res = "";
        int i = 0;
        while (i < _runeLen(s)) {
            String ch = s.substring(i, i+1);
            int j = 0;
            boolean found = false;
            while (j < 26) {
                if ((ch.equals(LOWER.substring(j, j+1)))) {
                    res = res + UPPER.substring(j, j+1);
                    found = true;
                    break;
                }
                j = j + 1;
            }
            if (found == false) {
                res = res + ch;
            }
            i = i + 1;
        }
        return res;
    }

    static boolean contains(String[] xs, String x) {
        int i_1 = 0;
        while (i_1 < xs.length) {
            if ((xs[i_1].equals(x))) {
                return true;
            }
            i_1 = i_1 + 1;
        }
        return false;
    }

    static boolean contains_char(String s, String ch) {
        int i_2 = 0;
        while (i_2 < _runeLen(s)) {
            if ((s.substring(i_2, i_2+1).equals(ch))) {
                return true;
            }
            i_2 = i_2 + 1;
        }
        return false;
    }

    static String get_value(String[] keys, String[] values, String key) {
        int i_3 = 0;
        while (i_3 < keys.length) {
            if ((keys[i_3].equals(key))) {
                return values[i_3];
            }
            i_3 = i_3 + 1;
        }
        return null;
    }

    static void print_mapping(String[] keys, String[] values) {
        String s = "{";
        int i_4 = 0;
        while (i_4 < keys.length) {
            s = s + "'" + keys[i_4] + "': '" + values[i_4] + "'";
            if (i_4 + 1 < keys.length) {
                s = s + ", ";
            }
            i_4 = i_4 + 1;
        }
        s = s + "}";
        System.out.println(s);
    }

    static String mixed_keyword(String keyword, String plaintext, boolean verbose) {
        String alphabet = UPPER;
        String keyword_u = String.valueOf(to_upper(keyword));
        String plaintext_u = String.valueOf(to_upper(plaintext));
        String[] unique = ((String[])(new String[]{}));
        int i_5 = 0;
        while (i_5 < _runeLen(keyword_u)) {
            String ch_1 = keyword_u.substring(i_5, i_5+1);
            if (((Boolean)(contains_char(alphabet, ch_1))) && contains(((String[])(unique)), ch_1) == false) {
                unique = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(unique), java.util.stream.Stream.of(ch_1)).toArray(String[]::new)));
            }
            i_5 = i_5 + 1;
        }
        int num_unique = unique.length;
        String[] shifted = ((String[])(new String[]{}));
        i_5 = 0;
        while (i_5 < unique.length) {
            shifted = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(shifted), java.util.stream.Stream.of(unique[i_5])).toArray(String[]::new)));
            i_5 = i_5 + 1;
        }
        i_5 = 0;
        while (i_5 < _runeLen(alphabet)) {
            String ch_2 = alphabet.substring(i_5, i_5+1);
            if (contains(((String[])(unique)), ch_2) == false) {
                shifted = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(shifted), java.util.stream.Stream.of(ch_2)).toArray(String[]::new)));
            }
            i_5 = i_5 + 1;
        }
        String[][] modified = ((String[][])(new String[][]{}));
        int k = 0;
        while (k < shifted.length) {
            String[] row = ((String[])(new String[]{}));
            int r = 0;
            while (r < num_unique && k + r < shifted.length) {
                row = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(row), java.util.stream.Stream.of(shifted[k + r])).toArray(String[]::new)));
                r = r + 1;
            }
            modified = ((String[][])(appendObj(modified, row)));
            k = k + num_unique;
        }
        String[] keys = ((String[])(new String[]{}));
        String[] values = ((String[])(new String[]{}));
        int column = 0;
        int letter_index = 0;
        while (column < num_unique) {
            int row_idx = 0;
            while (row_idx < modified.length) {
                String[] row_1 = ((String[])(modified[row_idx]));
                if (row_1.length <= column) {
                    break;
                }
                keys = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(keys), java.util.stream.Stream.of(alphabet.substring(letter_index, letter_index+1))).toArray(String[]::new)));
                values = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(values), java.util.stream.Stream.of(row_1[column])).toArray(String[]::new)));
                letter_index = letter_index + 1;
                row_idx = row_idx + 1;
            }
            column = column + 1;
        }
        if (((Boolean)(verbose))) {
            print_mapping(((String[])(keys)), ((String[])(values)));
        }
        String result = "";
        i_5 = 0;
        while (i_5 < _runeLen(plaintext_u)) {
            String ch_3 = plaintext_u.substring(i_5, i_5+1);
            String mapped = String.valueOf(get_value(((String[])(keys)), ((String[])(values)), ch_3));
            if ((mapped == null)) {
                result = result + ch_3;
            } else {
                result = result + mapped;
            }
            i_5 = i_5 + 1;
        }
        return result;
    }
    public static void main(String[] args) {
        UPPER = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        LOWER = "abcdefghijklmnopqrstuvwxyz";
        System.out.println(mixed_keyword("college", "UNIVERSITY", true));
    }

    static <T> T[] appendObj(T[] arr, T v) {
        T[] out = java.util.Arrays.copyOf(arr, arr.length + 1);
        out[arr.length] = v;
        return out;
    }

    static int _runeLen(String s) {
        return s.codePointCount(0, s.length());
    }
}

File: tests/algorithms/transpiler/FS/ciphers/mixed_keyword_cypher.fs
// Generated 2025-08-07 10:31 +0700

exception Break
exception Continue

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _idx (arr:'a array) (i:int) : 'a =
    if i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let UPPER: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
let LOWER: string = "abcdefghijklmnopqrstuvwxyz"
let rec to_upper (s: string) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable s = s
    try
        let mutable res: string = ""
        let mutable i: int = 0
        try
            while i < (String.length (s)) do
                try
                    let ch: string = string (s.[i])
                    let mutable j: int = 0
                    let mutable found: bool = false
                    try
                        while j < 26 do
                            try
                                if ch = (string (LOWER.[j])) then
                                    res <- res + (string (UPPER.[j]))
                                    found <- true
                                    raise Break
                                j <- j + 1
                            with
                            | Continue -> ()
                            | Break -> raise Break
                    with
                    | Break -> ()
                    | Continue -> ()
                    if found = false then
                        res <- res + ch
                    i <- i + 1
                with
                | Continue -> ()
                | Break -> raise Break
        with
        | Break -> ()
        | Continue -> ()
        __ret <- res
        raise Return
        __ret
    with
        | Return -> __ret
let rec contains (xs: string array) (x: string) =
    let mutable __ret : bool = Unchecked.defaultof<bool>
    let mutable xs = xs
    let mutable x = x
    try
        let mutable i: int = 0
        while i < (Seq.length (xs)) do
            if (_idx xs (i)) = x then
                __ret <- true
                raise Return
            i <- i + 1
        __ret <- false
        raise Return
        __ret
    with
        | Return -> __ret
let rec contains_char (s: string) (ch: string) =
    let mutable __ret : bool = Unchecked.defaultof<bool>
    let mutable s = s
    let mutable ch = ch
    try
        let mutable i: int = 0
        while i < (String.length (s)) do
            if (string (s.[i])) = ch then
                __ret <- true
                raise Return
            i <- i + 1
        __ret <- false
        raise Return
        __ret
    with
        | Return -> __ret
let rec get_value (keys: string array) (values: string array) (key: string) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable keys = keys
    let mutable values = values
    let mutable key = key
    try
        let mutable i: int = 0
        while i < (Seq.length (keys)) do
            if (_idx keys (i)) = key then
                __ret <- _idx values (i)
                raise Return
            i <- i + 1
        __ret <- unbox<string> null
        raise Return
        __ret
    with
        | Return -> __ret
let rec print_mapping (keys: string array) (values: string array) =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    let mutable keys = keys
    let mutable values = values
    try
        let mutable s: string = "{"
        let mutable i: int = 0
        while i < (Seq.length (keys)) do
            s <- ((((s + "'") + (_idx keys (i))) + "': '") + (_idx values (i))) + "'"
            if (i + 1) < (Seq.length (keys)) then
                s <- s + ", "
            i <- i + 1
        s <- s + "}"
        printfn "%s" (s)
        __ret
    with
        | Return -> __ret
let rec mixed_keyword (keyword: string) (plaintext: string) (verbose: bool) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable keyword = keyword
    let mutable plaintext = plaintext
    let mutable verbose = verbose
    try
        let alphabet: string = UPPER
        let keyword_u: string = to_upper (keyword)
        let plaintext_u: string = to_upper (plaintext)
        let mutable unique: string array = [||]
        let mutable i: int = 0
        while i < (String.length (keyword_u)) do
            let ch: string = string (keyword_u.[i])
            if (contains_char (alphabet) (ch)) && ((contains (unique) (ch)) = false) then
                unique <- Array.append unique [|ch|]
            i <- i + 1
        let num_unique: int = Seq.length (unique)
        let mutable shifted: string array = [||]
        i <- 0
        while i < (Seq.length (unique)) do
            shifted <- Array.append shifted [|_idx unique (i)|]
            i <- i + 1
        i <- 0
        while i < (String.length (alphabet)) do
            let ch: string = string (alphabet.[i])
            if (contains (unique) (ch)) = false then
                shifted <- Array.append shifted [|ch|]
            i <- i + 1
        let mutable modified: string array array = [||]
        let mutable k: int = 0
        while k < (Seq.length (shifted)) do
            let mutable row: string array = [||]
            let mutable r: int = 0
            while (r < num_unique) && ((k + r) < (Seq.length (shifted))) do
                row <- Array.append row [|_idx shifted (k + r)|]
                r <- r + 1
            modified <- Array.append modified [|row|]
            k <- k + num_unique
        let mutable keys: string array = [||]
        let mutable values: string array = [||]
        let mutable column: int = 0
        let mutable letter_index: int = 0
        try
            while column < num_unique do
                try
                    let mutable row_idx: int = 0
                    try
                        while row_idx < (Seq.length (modified)) do
                            try
                                let mutable row: string array = _idx modified (row_idx)
                                if (Seq.length (row)) <= column then
                                    raise Break
                                keys <- Array.append keys [|string (alphabet.[letter_index])|]
                                values <- Array.append values [|_idx row (column)|]
                                letter_index <- letter_index + 1
                                row_idx <- row_idx + 1
                            with
                            | Continue -> ()
                            | Break -> raise Break
                    with
                    | Break -> ()
                    | Continue -> ()
                    column <- column + 1
                with
                | Continue -> ()
                | Break -> raise Break
        with
        | Break -> ()
        | Continue -> ()
        if verbose then
            print_mapping (keys) (values)
        let mutable result: string = ""
        i <- 0
        while i < (String.length (plaintext_u)) do
            let ch: string = string (plaintext_u.[i])
            let mapped: string = get_value (keys) (values) (ch)
            if mapped = (unbox<string> null) then
                result <- result + ch
            else
                result <- result + mapped
            i <- i + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%s" (mixed_keyword ("college") ("UNIVERSITY") (true))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
